lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,7 +113,51 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deeplinkingFields, setDeeplinkingFields] = useState(
+    /** @type {Record<string,string>} */ null
+  );
+
+  const submit = useCallback(
+    async (/** @type {Content} */ content) => {
+      async function getDeeplinkingFields() {
+        const data = {
+          ...deeplinkingAPI.data,
+          content,
+          extra_params: {
+            groupSet: groupConfig.useGroupSet ? groupConfig.groupSet : null,
+          },
+        };
+        setDeeplinkingFields(
+          await apiCall({
+            authToken: deeplinkingAPI.authToken,
+            path: deeplinkingAPI.path,
+            data,
+          })
+        );
+      }
+      // When deeplinkingAPI is present we want to call the backend to return the form
+      // fields we'll forward to the LMS to complete the DeepLinking request
+      if (content && deeplinkingAPI && !deeplinkingFields) {
+        try {
+          await getDeeplinkingFields();
```

Comment:
What does the user see while this request is in-flight? ie. Do they see a loading state and are they prevented from changing anything or re-submitting while this request is in-flight? The `shouldSubmit` state causes a spinner to show, but that is not currently set until after this returns. One way to handle this might be to set the `shouldSubmit` state earlier, but change the `useEffect` that submits the form to check both for `shouldSubmit` being `true` and `deeplinkingFields` being set (if needed).

Reply:
Follow-up question: Does this new API modify any state on the backend? If not we could potentially call it eagerly while waiting for the user to confirm their content (that's out of scope for this PR).

Reply:
Updated following your suggestion (`useEffect that submits the form to check both for shouldSubmit being true and deeplinkingFields being set`)

> Follow-up question: Does this new API modify any state on the backend? If not we could potentially call it eagerly while waiting for the user to confirm their content (that's out of scope for this PR).

It doesn't change any state  at the moment but it might in the near future (ie record the assignment configuration in the database).

Not sure about eagerly doing this, it might complicate things around the JWT, double submissions, it might not be worth it.